File: app/channels/user_channel.rb
class UserChannel < ApplicationCable::Channel
  def subscribed
    @user = User.find_by(id: params[:user_id])

    @user ? stream_for(@user) : reject
  end

  def toggle
    new_status = @user.toggle_status

    UserChannel.broadcast_to(@user, status: new_status)
  end
end

File: app/javascript/channels/user_channel.js
import consumer from "./consumer";

export default function buildUserChannel(id, handleStatusUpdates) {
  return consumer.subscriptions.create(
    { channel: "UserChannel", user_id: id },
    {
      toggle(id) {
        this.perform("toggle", { user_id: id });
      },
      received: handleStatusUpdates,
    }
  );
}
